src/editor/StyleEditor.cpp

```diff
@@ -0,0 +1,136 @@
+#include "StyleEditor.h"
+
+#include <QApplication>
+#include <QDirIterator>
+#include <QFile>
+#include <QMessageBox>
+#include <QPalette>
+#include <QStyleFactory>
+
+#include "Settings.h"
+
+namespace StyleEditor {
+
+const QString requestedDefaultStyleName =
+    "ptCollage";  // This is the desired default style. In the event this style
+                  // is problematic, which is entirely possible, it will not be
+                  // re-set as the style; the System theme then will be used.
+                  // I don't see the need to have this change after
+                  // compile-time. The purpose of these styles is to be
+                  // configurable, and the user's preference is respected given
+                  // their respected style is not problematic. The name of this
+                  // style should be consistent, reliable, and guaranteed on a
+                  // fresh instance (installation or compilation). And if not,
+                  // there won't be any errors.
+
+void determineStyle() {
+  QString styleName = Settings::StyleName::get();
+  if (styleName == "System") return;
+  if (styleName.isEmpty() ||
+      !QFile::exists(qApp->applicationDirPath() + "/style/" + styleName + "/" +
```

Comment:
Would you mind refactoring this logic that gets the qss file of a stylesheet into a function maybe?

Reply:
done